x-pack/legacy/plugins/siem/server/lib/detection_engine/alerts/build_events_query.ts

```diff
@@ -58,25 +58,40 @@ export const buildEventsScrollQuery = ({
       },
     },
   ];
-  return {
+  const searchQuery = {
     allowNoIndices: true,
     index,
-    scroll,
     size,
     ignoreUnavailable: true,
     body: {
       query: {
         bool: {
           filter: [
-            ...filterWithTime,
+            ...[filterWithTime],
             {
               match_all: {},
             },
           ],
         },
       },
       track_total_hits: true,
-      sort: ['_doc'],
+      sort: [
+        {
+          '@timestamp': {
+            order: 'asc',
+          },
+        },
+      ],
     },
   };
+  if (searchAfterSortId) {
```

Comment:
This will return false for empty strings which might or might not be what you want instead of string checking of just say null/undefined with `if (searchAfter != null)`?

If you do want this as is what you want then I would add it to the test file along with the other test above I created like so which will give future maintainers something to think about if they change this from `if(searchAfterSortId != null)` as they will have to change the test to indicate that empty strings should or should not return a body with the search_after.

```ts
  test('if searchAfterSortId is an empty string it should not be included', () => {
    const query = buildEventsSearchQuery({
      index: ['auditbeat-*'],
      from: 'now-5m',
      to: 'today',
      filter: {},
      size: 100,
      searchAfterSortId: '',
    });
    expect(query).toEqual({
      allowNoIndices: true,
      index: ['auditbeat-*'],
      size: 100,
      ignoreUnavailable: true,
      body: {
        query: {
          bool: {
            filter: [
              [
                {},
                {
                  bool: {
                    filter: [
                      {
                        bool: {
                          should: [
                            {
                              range: {
                                '@timestamp': {
                                  gte: 'now-5m',
                                },
                              },
                            },
                          ],
                          minimum_should_match: 1,
                        },
                      },
                      {
                        bool: {
                          should: [
                            {
                              range: {
                                '@timestamp': {
                                  lte: 'today',
                                },
                              },
                            },
                          ],
                          minimum_should_match: 1,
                        },
                      },
                    ],
                  },
                },
              ],
              {
                match_all: {},
              },
            ],
          },
        },
        track_total_hits: true,
        sort: [
          {
            '@timestamp': {
              order: 'asc',
            },
          },
        ],
      },
    });
  });
```

If you do not intend for an empt

Reply:
I think I want it to equal false as I believe search_after will throw an error if an empty timestamp is passed as the sort id to search after on the next search. Just added the test. Thank you very much.